File: MonteSeuComputador 2.0/app/src/main/java/br/com/senaicimatec/monteseucomputador/adapter/ProductAdapter.java
package br.com.senaicimatec.monteseucomputador.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;
import br.com.senaicimatec.monteseucomputador.R;
import br.com.senaicimatec.monteseucomputador.model.Product;

public class ProductAdapter extends RecyclerView.Adapter<ProductAdapter.ProductViewHolder>  {
    List<Product> productList;

    public ProductAdapter(List<Product> productList){
        this.productList = productList;
    }

    @NonNull
    @Override
    public ProductViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View itemList = LayoutInflater.from(parent.getContext()).inflate(R.layout.produto_item, parent, false);
        return new ProductViewHolder(itemList);
    }

    @Override
    public void onBindViewHolder(@NonNull ProductViewHolder holder, int position) {
        holder.productPhoto.setImageResource(productList.get(position).getPhoto());
        holder.productName.setText(productList.get(position).getName());
        holder.productDescription.setText(productList.get(position).getDescription());
        holder.productPrice.setText(productList.get(position).getPrice());
    }

    @Override
    public int getItemCount() {
        return productList.size();
    }

    static class ProductViewHolder extends RecyclerView.ViewHolder{

        ImageView productPhoto = itemView.findViewById(R.id.productImage);
        TextView productName = itemView.findViewById(R.id.name);
        TextView productDescription = itemView.findViewById(R.id.productDescription);
        TextView productPrice = itemView.findViewById(R.id.productPrice);

        public ProductViewHolder(@NonNull View itemView) {
            super(itemView);
        }
    }
}

File: BuildYourComputer/app/src/main/java/com/example/buildyourcomputer/adapter/ProductAdapter.kt
package com.example.buildyourcomputer.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.buildyourcomputer.Product
import com.example.buildyourcomputer.R
import java.text.NumberFormat
import java.util.*
import kotlin.collections.ArrayList

class ProductAdapter(productList: List<Product>) :
    RecyclerView.Adapter<ProductAdapter.ProductViewHolder>() {
    var productList: List<Product> = ArrayList()

    init {
        this.productList = productList
    }

    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): ProductViewHolder {
        val listItem: View =
            LayoutInflater.from(parent.context).inflate(R.layout.product_item, parent, false)
        return ProductViewHolder(listItem)
    }

    inner class ProductViewHolder(itemView: View) :
        RecyclerView.ViewHolder(itemView) {
        var productImage: ImageView = itemView.findViewById(R.id.product_image)
        var productName: TextView = itemView.findViewById(R.id.product_name)
        var productDescription: TextView = itemView.findViewById(R.id.product_description)
        var productPrice: TextView = itemView.findViewById(R.id.preco_produto)
    }

    override fun onBindViewHolder(holder: ProductViewHolder, position: Int) {
        val format: NumberFormat = NumberFormat.getCurrencyInstance()
        format.maximumFractionDigits = 0
        format.currency = Currency.getInstance("BRL")

        holder.productImage.setImageResource(productList[position].image)
        holder.productName.text = productList[position].name
        holder.productDescription.text = productList[position].description
        holder.productPrice.text = format.format(productList[position].price)
    }

    override fun getItemCount(): Int {
        return productList.size
    }
}